model/src/cfd.rs

```diff
@@ -918,11 +918,15 @@ impl Cfd {
         proposal: SettlementProposal,
         sig_taker: Signature,
     ) -> Result<CollaborativeSettlement> {
+        anyhow::ensure!(
+            self.role == Role::Maker,
+            "Only the maker can create complete collaborative settlement signing"
+        );
```

Comment:
I'd go even further by using `assert_debug` or `assert` perhaps - it's not a runtime error, it's a bug in our code if this ever happens, therefore I'd avoid gracefully handling this.

I'm happy also to settle on this one for the time being as it's a giant improvement guarding us from actually doing it when we shouldn't

Reply:
I think you're right. I went for `debug_assert`.